jwst/pipeline/calwebb_spec3.py

```diff
@@ -148,7 +149,7 @@ def process(self, input):
 
             # If the step is skipped, do the container splitting that
             # would've been done in master_background
-            if self.master_background.skip:
+            if query_step_status(source_models, "master_background") == 'SKIPPED':
```

Comment:
```suggestion
            if set(query_step_status(source_models, "master_background")) == {SKIPPED}:
```
This looks to be the only usage of `query_step_status` that uses a `ModelContainer` as input. Are there any others that I missed? Since we know here that `source_models` is a container the above suggestion (if `query_step_status` returns a list of statuses) would check that all `source_models` have their `master_background` set to "SKIPPED" (an import for this constant is also needed).

Reply:
And my understanding of https://github.com/spacetelescope/jwst/pull/8600#discussion_r1658742018 (correct me if I'm wrong) is that this could also be replaced by `query_step_status(source_models[0], "master_background" == SKIPPED)`. Is that right?

Reply:
Looking at the logic for the master background step, yes, it looks like that's true in this case.  It's only ever skipped if it's inappropriate to run the step on any of the sub-models.

I was thinking of steps like wavecorr, where individual slits within a MultiSlitModel might be wavelength corrected or not, depending on their source type.  It seems like it would be useful to have the query_step_status function return something useful in that case too: either the step ran usefully on at least one model in the container or it did not. 

[JP-3630](https://jira.stsci.edu/browse/JP-3630) is relevant here: we do not currently add step status to individual slits in MultiSlitModels everywhere it's relevant but it would be useful to have that information.  If you prefer, we can defer this discussion until that ticket gets picked up.

Reply:
Thanks! I think that would require additional logic since it appears `MultiSlitModel` is not a `Sequence` (and only the "top" level `cal_step` status would be considered). Thanks for sharing the ticket. It mentions that only reporting if any slit was "COMPLETE" is less helpful than providing details about which slits are "COMPLETE". This seems similar to `query_step_status` returning a list of statuses for a container vs having it report the result of an `any` (or `all`) combination of the statuses.

Given that the changes in this PR don't seem to require `query_step_status` and that there are other considerations (like MultiSlitModel). I suggest we remove `query_step_status` from this PR since the main goal is removing `Step.skip` and that seems doable without adding this new API.

Reply:
The complication is that the MasterBackground step is sometimes passed a ModelContainer and sometimes not.  This was the original reason I split `query_step_status` into its own function.  I have now taken the suggestion to make the `query_step_status` in stpipe accept a datamodel, and then put the `if isinstance(model, ModelContainer)...` logic into the MasterBackground step.  Hopefully this sounds reasonable and makes any additional changes for JP-3630 easy enough to make.

Reply:
Also, this is not the only instance of calling `query_step_status` on a ModelContainer - there are also two inside `calwebb_spec2`.  My reading of this step suggests that checking the zeroth-index model is ok here too, but I could be wrong...

Reply:
The same is true of the instances in calwebb_spec2: sometimes they receive a ModelContainer, and sometimes not.  I guess for now I'll add the logic into both places

Reply:
After some more discussion with Melanie, we agreed to revert this to its original form, where only the zeroth index is checked for a ModelContainer.  Any changes to the `query_step_status()` function that are needed in the future can be handled then.